docs/DevGuide/Versioning.md

```diff
@@ -0,0 +1,117 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+
+# Versioning and releases {#dev_guide_versioning}
+
+\tableofcontents
+
+# Version format
+
+We employ a date-based versioning scheme with the format `YYYY.0M.0D[.TWEAK]`,
+where `YYYY` is the year, `0M` the zero-padded month and `0D` the zero-padded
+day of the month when the version is released. If additional versions are
+released on a single day, the `TWEAK` component enumerates them. For example,
+the first version released on May 4th, 2021 carries the name `2021.05.04`.
+Additional versions released on May 4th, 2021 would carry the names
+`2021.05.04.1`, `2021.05.04.2`, etc.
+
+The reason we use a date-based versioning scheme over, for example, semantic
+versioning is to avoid implying a notion of compatibility between releases (see
+section below).
+
+# Guarantees and non-guarantees of releases
```

Comment:
We don't really have a good rule for what goes in the devguide and what goes elsewhere, but I think the devguide is intended for at least somewhat technical and somewhat internal information.  This is about as far as we're ever likely to get from that.  (It is addressed to *users*.)

Reply:
Agreed. It's not a user tutorial either though.. do you have an idea where to put it? Or should it be split somehow?

Reply:
I think a split makes sense.  Everything from "Automatic versioning" downwards is really technical, some of it only usable by core devs or something like that.  That seems like a dev guide entry.  The stuff above that is aimed at users.  Maybe link it from the installation page?